e2e-tests/micronaut-rest-api/src/main/java/com/foo/micronaut/rest/MicronautApplication.java

```diff
@@ -0,0 +1,49 @@
+package com.foo.micronaut.rest;
+
+import io.micronaut.runtime.Micronaut;
+import io.micronaut.context.ApplicationContext;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.info.Info;
+
+@OpenAPIDefinition(
+        info = @Info(
+                title = "micronaut",
+                version = "latest",
+                description = "Micronaut E2E Test API"))
+public class MicronautApplication {
+
+    private ApplicationContext context;
+
+    public MicronautApplication() {}
+
+    public static void main(String[] args) {
+
+        try {
+            new MicronautApplication().run();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void run() {
```

Comment:
the property `context` and the methods (such as `run`, `getPort`, `isRunning`, and `stop`) might better move to Controller.

Reply:
As we discussed, leaving it as it is for now